app/code/Magento/InventoryShipping/view/adminhtml/web/js/form/form.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([
+    'jquery',
+    'Magento_Ui/js/form/form',
+    'mageUtils',
+    'uiRegistry'
+], function ($, Form, utils, registry) {
+    'use strict';
+
+    return Form.extend({
+        defaults: {
+            sourceSelectionUrl: '${ $.sourceSelectionUrl }'
+        },
+
+        /**
+         * Validate and save form.
+         *
+         * @param {String} redirect
+         * @param {Object} data
+         */
+        processAlgorithm: function (redirect, data) {
+            var process = $.Deferred(),
```

Comment:
$.ajax call also returns Deffered object, no reason create another Promise and resolve in inside success callback.